lib/pbench/server/api/resources/upload_api.py

```diff
@@ -50,70 +51,100 @@ def __init__(self, config, logger):
         )
 
     @Auth.token_auth.login_required()
-    def put(self, controller):
-
+    def put(self, filename_arg):
         try:
             username = Auth.token_auth.current_user().username
         except Exception:
-            self.logger.exception("Tarfile upload: Exception verifying the username")
+            self.logger.exception("Error verifying the username")
             abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
 
-        if not request.headers.get("filename"):
-            self.logger.debug(
-                "Tarfile upload: Post operation failed due to missing filename header"
+        filename = secure_filename(filename_arg)
+        if not self.allowed_file(filename):
+            self.logger.warning(
+                "Unsupported file extension for {}, user = {}", filename_arg, username
             )
             abort(
-                400,
-                message="Missing filename header, POST operation requires a filename header to name the uploaded file",
+                HTTPStatus.BAD_REQUEST,
+                message="File extension not supported, must be"
+                f" {self.ALLOWED_EXTENSION}",
             )
-        filename = secure_filename(request.headers.get("filename"))
 
-        if not request.headers.get("Content-MD5"):
-            self.logger.debug(
-                f"Tarfile upload: Post operation failed due to missing md5sum header for file {filename}"
+        controller_arg = request.headers.get("controller")
+        if not controller_arg:
+            msg = "Missing required controller header"
+            self.logger.warning(
+                "{} for user = {}, file = {}", msg, username, filename_arg
             )
-            abort(
-                400,
-                message="Missing md5sum header, POST operation requires md5sum of an uploaded file in header",
-            )
-        md5sum = request.headers.get("Content-MD5")
+            abort(HTTPStatus.BAD_REQUEST, message=msg)
```

Comment:
Repeated pattern here for all the required headers; this could be a helper method that takes the header name and returns the value.

Reply:
Yes, please.  However, I think it will need to take more than just the header name -- it'll presumably want the `msg`, `username`, `filename_arg`, and HTTP status code, as well.  :-/  And, I'm not sure how I feel about burying the `abort()` call inside it -- that should probably be done at the top level.  :-(

And, this call is different from the subsequent ones, as it omits the controller from the message (although, that could be finessed).

Reply:
I agree it is a repeated pattern, but it won't save us much with all the arguments that have to be provided.